app/controllers/aid_applications/edits_controller.rb

```diff
@@ -21,6 +25,8 @@ def update
       respond_with @aid_application, location: (lambda do
         if @aid_application.submitted?
           edit_organization_aid_application_verification_path(current_organization, @aid_application)
+        elsif @aid_application.allow_mailing_address == true
```

Comment:
You should do this off the form action, otherwise it will _always_ redirect back to this same page on form submit